src/Transports.AspNetCore/AuthorizationVisitor.cs

```diff
@@ -0,0 +1,37 @@
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <inheritdoc/>
+public class AuthorizationVisitor : AuthorizationVisitorBase
+{
+    /// <inheritdoc cref="AuthorizationVisitor"/>
+    public AuthorizationVisitor(ValidationContext context, ClaimsPrincipal claimsPrincipal, IAuthorizationService authorizationService)
+        : base(context)
+    {
+        if (context == null)
+            throw new ArgumentNullException(nameof(context));
+        ClaimsPrincipal = claimsPrincipal ?? throw new ArgumentNullException(nameof(claimsPrincipal));
+        AuthorizationService = authorizationService ?? throw new ArgumentNullException(nameof(authorizationService));
+    }
+
+    /// <summary>
+    /// Gets the user that this authorization visitor will authenticate against.
+    /// </summary>
+    protected ClaimsPrincipal ClaimsPrincipal { get; }
+
+    /// <summary>
+    /// Gets the authorization service that is used to authorize policy requests.
+    /// </summary>
+    protected IAuthorizationService AuthorizationService { get; }
+
+    /// <inheritdoc/>
+    protected override bool Authorize()
```

Comment:
Method looks like an action to perform (verb). The same for AuthorizeRole. Do they definitely have such names?

Reply:
I named them so they all start with `Authorize`, and so they are similar to the extension methods we have on defined within GraphQL.NET, which are `Authorize`, `AuthorizeWithRoles` and `AuthorizeWithPolicy`.  We could name them similar to the members they map to:

- `bool IsAuthenticated { get; }`,
- `bool IsInRole(string role)`, and
- `AuthorizationResult Authorize(string policy)`

What do you think?  I'm open to suggestions and would like to do whatever makes the most sense.

For reference here is the code:

```cs
protected override bool Authorize()
    => ClaimsPrincipal.Identity?.IsAuthenticated ?? false;

protected override bool AuthorizeRole(string role)
    => ClaimsPrincipal.IsInRole(role);

protected override AuthorizationResult AuthorizePolicy(string policy)
    => AuthorizationService.AuthorizeAsync(ClaimsPrincipal, policy).GetAwaiter().GetResult();
```

Reply:
It is bad that for all of them you can not use the name `IsXXX`. The last one method is verb. There is no 100% symmetry. I don't know how to do better.


    bool IsAuthenticated { get; },
    bool IsInRole(string role), and
    AuthorizationResult Authorize(string policy)

This variant at least looks reasonable.

Reply:
Will do.  At least it matches the methods they call (by default).